tests/testcases/015_check-pods-running.yml

```diff
@@ -0,0 +1,37 @@
+---
+- hosts: kube-master[0]
+  tasks:
+
+  - name: Force binaries directory for Container Linux by CoreOS
+    set_fact:
+      bin_dir: "/opt/bin"
+    when: ansible_os_family in ["CoreOS", "Container Linux by CoreOS"]
+
+  - set_fact:
+      bin_dir: "/usr/local/bin"
+    when: not ansible_os_family in ["CoreOS", "Container Linux by CoreOS"]
+
+  - name: Check kubectl output
+    shell: "{{bin_dir}}/kubectl get pods --all-namespaces -owide"
+    register: get_pods
+    no_log: true
+
+  - debug: msg="{{get_pods.stdout.split('\n')}}"
+
+  - name: Check that all pods are running
+    shell: "{{bin_dir}}/kubectl get pods --all-namespaces --no-headers -o yaml"
+    register: run_pods_log
+    until:
+    - '(run_pods_log.stdout | from_yaml)["items"] | map(attribute = "status.phase") | unique == ["Running"]'
```

Comment:
change this to:

`- ( (run_pods_log.stdout | from_yaml)['items'] | map(attribute = 'status.phase')  | unique | first ) == 'Running'`

And it will work

Reply:
I've been testing this. If `status.phase` is `[Running, Pending]` then `| unique | first == 'Running'` will pass but there is still a pending container.

Reply:
Thats because using `| unique' on `[Running, Pending]`  will give you a set of only `[Running, Pending]` which is sorted with Running first. And since you then use `| first' to take the first element, it passes. Then you need to check the size of that set is 1 and only contains Running

Reply:
I resolved this by using `| list == ["Running"]`